src/room17/SkyBlock/SkyBlockListener.php

```diff
@@ -226,10 +237,17 @@ public function onQuit(PlayerQuitEvent $event): void {
             $isle->tryToClose();
         }
     }
+    
     /**
-     * @param PlayerQuitEvent $event
+     * @param PlayerBedEnterEvent $event
      */
-    public function onSleep(PlayerBedEnterEvent $event){
-        $event->setCancelled(true);
+    public function onSleep(PlayerBedEnterEvent $event): void {
+        $player = $event->getPlayer();
+        $session = $this->getSession($player);
```

Comment:
This makes more sense now, although this line doesn't make sense at all. Why would you get the session if you're not going to use it?